firelink/apps.py

```diff
@@ -51,6 +51,43 @@ def _process_bonfire_apps_list(self, apps):
         apps_array.sort(key=lambda app: app['name'])
         return apps_array
 
+    def get_processed_template(self, request):
+        """Get the processed template for an app."""
+        namespace = request["namespace"]
+        clowd_env = self._get_clowdenv_for_ns(namespace)
+        if not clowd_env:
+            raise bonfire.FatalError("Could not find a ClowdEnvironment tied to ns '{ns}'.")
+        request["clowd_env"] = clowd_env
+        apps_config = bonfire._process(
+            request["app_names"],
+            request["source"],
+            request["get_dependencies"],
+            request["optional_deps_method"],
+            request["local_config_method"],
+            request["set_image_tag"],
+            request["ref_env"],
+            request["fallback_ref_env"],
+            request["target_env"],
+            request["set_template_ref"],
+            request["set_parameter"],
+            request["clowd_env"],
+            request["local_config_path"],
+            AppOrComponentSelector(bool(request.get("remove_resources")), request.get("remove_resources", []), []),
+            AppOrComponentSelector(bool(request["no_remove_resources"]), request.get("no_remove_resources", []), []),
+            AppOrComponentSelector(bool(request["remove_dependencies"]), [], request.get("remove_dependencies",[])),
+            AppOrComponentSelector(bool(request["no_remove_dependencies"]), [], request.get("no_remove_dependencies", [])),
```

Comment:
Same issue with these ones, only additional thing to point out is you'll hit a `KeyError` on this if "no_remove_resources" is not present: `bool(request["no_remove_resources"])

A safer way would be `bool(request.get("no_remove_resources"))` ... I think this is all moot though because of my comment above :)

Reply:
Pushed up a revision that addresses this part